src/generate.js

```diff
@@ -2,39 +2,89 @@ import Prefixer from 'inline-style-prefixer';
 
 import {
     objectToPairs, kebabifyStyleName, recursiveMerge, stringifyValue,
-    importantify
+    importantify,
 } from './util';
 
+// In a bundle of styles, find all of the different class names that a single
+// named descendant selector can have.
+const findNamesForDescendants = (styles, names) => {
+    Object.keys(styles).forEach(key => {
+        if (key[0] === ':' || key[0] === '@') {
+            // Recurse for pseudo or @media styles
+            findNamesForDescendants(styles[key], names);
+        } else if (key[0] === '>' && key[1] === '>') {
+            // Recurse for descendant styles
+            findNamesForDescendants(styles[key], names);
+
+            // Pluck out all of the names in the _names object.
+            Object.keys(styles[key]._names).forEach(name => {
+                names[key] = names[key] || [];
+                names[key].push(name);
+            });
+        }
+    });
+};
+
 export const generateCSS = (selector, styleTypes, stringHandlers,
-        useImportant) => {
+                            useImportant) => {
     const merged = styleTypes.reduce(recursiveMerge);
 
+    const classNamesForDescendant = {};
+    findNamesForDescendants(merged, classNamesForDescendant);
+
+    return generateCSSInner(
+        selector, merged, stringHandlers, useImportant,
+        classNamesForDescendant);
+}
+
+const generateCSSInner = (selector, style, stringHandlers,
```

Comment:
This is getting pretty complicated, and could also be greatly aided by a JSdoc, IMO.